### PR TITLE
Add `find_sc` notebooks for LM and UMAP using Plate 4 data

### PR DESCRIPTION
**Pull request by jenna-tomkinson**
# Add `find_sc` notebooks for LM and UMAP using Plate 4 data

In this PR, I add notebooks to the `linear_model` and `UMAP` modules in the analyze data folder to find and generate single-cell crops based on either the top single-cells per top linear model feature or the UMAP cluster.

### CODE REVIEW

File: 4.analyze_data/notebooks/UMAP/nbconverted/1.visualize-UMAP-plate4.r [hunk 1]
```diff
@@ -165,4 +166,59 @@ ggsave(output_file, umap_cell_type_figure, dpi = 500, height = 10, width = 10)
...
+    axis.text = element_text(size = 12),
+    
+    # Increase legend size
+    legend.title = element_text(size = 14),
+    legend.text = element_text(size = 14),
+  ) +
+  coord_fixed(1.1)
+
+# saving image
+output_file <- paste0(output_umap_file, "_number_neighbors_cells.png")
```
Comment (MikeLippincott) [inline 4.analyze_data/notebooks/UMAP/nbconverted/1.visualize-UMAP-plate4.r:219]: consider using the file.path function here for interoperability
 ```suggestion
output_file <- file.path(paste0(output_umap_file, "_number_neighbors_cells.png"))
```

File: 4.analyze_data/notebooks/UMAP/nbconverted/2.plate4_find_sc.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+output_dir = pathlib.Path("plate4_umap_crops")
+output_dir.mkdir(parents=True,exist_ok=True)
+
+# Output dir for composite and cropped images
+output_img_dir = pathlib.Path("./images")
+output_img_dir.mkdir(exist_ok=True)
+
+
+# ## Find single-cells per cluster
```
Comment (MikeLippincott) [inline 4.analyze_data/notebooks/UMAP/nbconverted/2.plate4_find_sc.py:36]: consider removing vestigial code

**Review by MikeLippincott — APPROVED** [review]
I think this PR would benefit from a README explaining the difference between the linear modeling sc images and the UMAP sc images. Also, if there could be an intermediate file that stores the top N features from the LM and that is read into the script, this would be a safer way to get these images.


I would also consider making the bounding box slightly larger. These are all suggestions. Other than the comments I made this PR looks ready to merge! LGTM.

File: 4.analyze_data/notebooks/UMAP/nbconverted/1.visualize-UMAP-plate4.r (hunk 1 above)
Comment (jenna-tomkinson) [reply 4.analyze_data/notebooks/UMAP/nbconverted/1.visualize-UMAP-plate4.r:219]: Thank you for bringing this up! I didn't realize that was a thing haha. I have added this!

File: 4.analyze_data/notebooks/UMAP/nbconverted/2.plate4_find_sc.py (hunk 2 above)
Comment (jenna-tomkinson) [reply 4.analyze_data/notebooks/UMAP/nbconverted/2.plate4_find_sc.py:36]: I am not sure the code that you are referring to, but I will make sure to remove any unused code!

**Comment by jenna-tomkinson** [comment]
@MikeLippincott Thank you lots for the review! I implemented a README like you suggested! I think it helps explain a lot more what is going on.

I will be merging now!

**Pull request merged**